routes/analytics.js

```diff
@@ -88,20 +88,31 @@ function analytics(req, advertiserChannels, skipPublisherFiltering) {
 		skipPublisherFiltering
 	)
 	const appliedLimit = Math.min(200, limit)
+
+	const group = {
+		_id: {
+			$subtract: [{ $toLong: '$created' }, { $mod: [{ $toLong: '$created' }, span] }]
+		},
+		value: { $sum: '$value' }
+	}
+	const resultProjection = { value: '$value', time: '$_id', _id: 0 }
+	// checks if publisher requests result to be segmented
+	// by channel
+	if (req.query.segmentByChannel && !skipPublisherFiltering) {
+		project.channelId = 1
```

Comment:
to avoid mutating `project` here, you can simply always return it from `getProjAndMatch` - this does not cause harm/overhead in the non-segmented case